python/cuda_parallel/cuda/parallel/experimental/_utils/protocols.py

```diff
@@ -48,7 +57,7 @@ def is_contiguous(arr: DeviceArrayLike) -> bool:
     if strides is None:
         return True
 
-    shape = cai["shape"]
+    shape = get_shape(arr)
```

Comment:
We were reusing the dictionary produced by evaluation of `__cuda_array_interface__` property (relatively expensive to compute), but with this change we are effectively throwing the known result away and rebuilding the dictionary again in the call to `get_shape` function.

Reply:
I made this change to be consistent, but you make a good point so I reverted it